code cleanup and reorganization

mikeLitster opened the pull request:
break multiconglomerate function BL_program up into smaller functions

mikeLitster committed:
code cleanup and reorganization

break multiconglomerate function BL_program up into smaller functions

ToddR85 commented on Gen6_BL_Demo/Gen6_BL.cpp:
```diff
@@ -6,53 +6,61 @@
+#ifdef __cplusplus
+extern "C" {
+#endif
+
```
This is a CPP file so __cplusplus is not necessary although it doesn't hurt anything to have it here.

mikeLitster replied:
Moved these to the header files.

cirque-jonb commented on Gen6_BL_Demo/Borax_BL.c:
```diff
@@ -25,421 +25,537 @@
+    }
 
-  I2C_endTransmission(true);
+    I2C_write_DummyBytes(521 - numBytes);
```
521 is 520 + 1, 520 is the MAX_DATA_PAYLOAD_SIZE. This feels like it should be some define-based value
MAX_DATA_PAYLOAD_INDEX_LIMIT maybe?
#define MAX_DATA_PAYLOAD_INDEX_LIMIT MAX_DATA_PAYLOAD_SIZE + 1

mikeLitster replied:
I made a lot of new #defines

ToddR85 commented on Gen6_BL_Demo/Gen6_BL.cpp:
```diff
@@ -6,53 +6,61 @@
-	Serial.printf("Version: 0x%02X\n", status.Version);
-	Serial.printf("Error: 0x%02X\n", status.LastError);
+    Serial.printf("Sentinel: 0x%04X\n", status.Sentinel);
+    Serial.printf("Version: 0x%02X\n", status.Version);
```
I wonder if we should wrap all Serial.printf statements with a #define to make it easier for customers to swap out the print statement with their own and to make it easier to turn the printf statements on / off. 

Something like this:
    #define VA_ARGS(...) , ##__VA_ARGS__
    #define LOGF(Msg, ...) {Serial.printf(Msg VA_ARGS(__VA_ARGS__));}

mikeLitster replied:
It doesn't really make sense to turn them off, since it is a console program.

ToddR85 replied:
Customers use this code as a baseline for writing their own BL host so I was just thinking that wrapping the printf statements will make it easier to port this code to a different platform. Again, not a big deal, just a thought.

cirque-jonb commented on Gen6_BL_Demo/Borax_BL.c:
```diff
@@ -25,421 +25,537 @@
+#define MAX_DATA_PAYLOAD_SIZE 520
+
 static uint8_t _i2cSlaveAddr;
 static uint8_t _dataBuffer[533];
```
This feels like the 533 should be a define as well, MAX_DATA_PAYLOAD_SIZE + COMMAND_OVERHEAD_SIZE or something like that

cirque-jonb commented on Gen6_BL_Demo/Borax_BL.c:
```diff
@@ -25,421 +25,537 @@
+    I2C_beginTransmission(_i2cSlaveAddr);
+    I2C_write_SetFeatureReport(CMD_VALIDATE);
+    I2C_write(validationType);
+    I2C_write_DummyBytes(530 - 2);
```
I guess looking over Dave's spreadsheet and getting a set of defines that nail down 520, 521, 530-2, 533 and having it in a commented block that helps define the packet structure would be helpful

ToddR85 commented on Gen6_BL_Demo/Gen6_BL_Demo.ino:
```diff
@@ -2,75 +2,77 @@
 #include "Gen6_BL.h"
 #include "Borax_BL.h"
+#include <i2c_t3.h>
 #include "I2C.h"
```
Why is #include <i3c_t3.h> required here?  It probably should be included inside of I2c.h.  Note: If you do add it to I2c.h then it can be removed from I2c.cpp

mikeLitster replied:
Done

ToddR85 commented on Gen6_BL_Demo/Fletcher32.h:
```diff
@@ -35,7 +35,7 @@
  */
-extern uint32_t Fletcher32(void const *ptr, uint32_t n_bytes);
+uint32_t Fletcher32(void const *ptr, uint32_t n_bytes);
 
```
I agree with the change here but I would probably leave this an extern because I believe that this file is just copied from borax so it it better that it matches borax.  Not a big deal but something to consider.

mikeLitster replied:
reverted back to extern

ToddR85 reviewed (commented):
Mike, have you finished the UTF test for this code yet?  I would hold off on merging this code back into master until the test has been written and everything is passing

cirque-jonb commented on Gen6_BL_Demo/Borax_BL.c:
```diff
@@ -25,421 +25,537 @@
+}
 
-  I2C_write_GetFeatureReport();
+/* NOTE:
```
Love it.

cirque-jonb commented on Gen6_BL_Demo/Borax_BL.c:
```diff
@@ -25,421 +25,537 @@
+    uint16_t TargetHIDDescAddr;
+    uint32_t EntryPoint;
+
+    if(status->Version >= 0x08)
```
I wonder - do we just ignore versions 0..9 and only support versions 10 and 11?
Versions 8 and 9 are Gen4.
If we ignore 8 and 9 it simplifies things a tiny bit.

mikeLitster replied:
We are no longer supporting anything before 10

mikeLitster committed:
since all packets contain status, revamped code to reflect this

split I2C code into separate file to simplify code
replaced numbers with #defines for packet sizes
moved extern C guard statements to header files

mikeLitster committed:
I2C commands moved to Borax_BL_I2C_Commands.[ch]

mikeLitster committed:
Streamline BL_program so it reads well, add error printfs to arduino sketch
